chemprop/nn/predictors.py

```diff
@@ -191,24 +189,19 @@ class EvidentialFFN(RegressionFFN):
     _T_default_criterion = EvidentialLoss
 
     def forward(self, Z: Tensor) -> Tensor:
-        Y = super().forward(Z)
-        mean, v, alpha, beta = torch.chunk(Y, self.n_targets, 1)
-
-        mean = self.scale * mean + self.loc
-        v = v * self.scale**2
-
-        return torch.cat((mean, v, alpha, beta), 1)
-
-    def train_step(self, Z: Tensor) -> Tensor:
-        Y = super().forward(Z)
+        Y = self.ffn(Z)
         mean, v, alpha, beta = torch.chunk(Y, self.n_targets, 1)
-
         v = F.softplus(v)
         alpha = F.softplus(alpha) + 1
         beta = F.softplus(beta)
 
+        mean = self.output_transform(mean)
+        beta = self.output_transform.transform_variance(beta)
```

Comment:
Why is the scaling applied to beta here, but it was applied to v before?

Reply:
The variance is calculated as $\frac{{\beta}}{\lambda(\alpha-1)}$. Since the variance is proportional to $\beta$, the output transform should be done on it.

Reply:
Is $\lambda$ the same as v here in the code?

Reply:
Yes!